src/FeeRouter/ParentToChildRewardRouter.sol

```diff
@@ -93,4 +112,59 @@ contract ParentToChildRewardRouter is DistributionInterval {
         });
         emit FundsRouted(amount);
     }
+
+    /// @notice send full token balance in this contract to destination. Uses sender's address for fee refund
+    /// @param parentChainTokenAddr todor
+    /// @param maxSubmissionCost submission cost for retryable ticket
+    /// @param gasLimit gas limit for l2 execution of retryable ticket
+    /// @param maxFeePerGas max gas l2 gas price for retryable ticket
+    function routeToken(address parentChainTokenAddr, uint256 maxSubmissionCost, uint256 gasLimit, uint256 maxFeePerGas)
+        public
+        payable
+    {
+        // use routeNativeFunds for native currency, not this method
+        if (parentChainTokenAddr == NATIVE_CURRENCY) {
+            revert WrongMethod();
+        }
+        if (!canDistribute(parentChainTokenAddr)) {
+            revert DistributionTooSoon(block.timestamp, nextDistributions[parentChainTokenAddr]);
+        }
+
+        if (gasLimit < minGasLimit) {
+            revert GasLimitTooLow(gasLimit);
+        }
+
+        if (maxFeePerGas < minGasPrice) {
+            revert GasPriceTooLow(maxFeePerGas);
+        }
+
+        uint256 amount = IERC20(parentChainTokenAddr).balanceOf(address(this));
+        if (amount == 0) {
+            revert NoFundsToDistrubute();
+        }
+        // get gateway from gateway router
+        address gateway = parentChainGatewayRouter.getGateway(address(parentChainTokenAddr));
+        // approve amount on gateway
+        IERC20(parentChainTokenAddr).approve(gateway, amount);
+
+        // encode max submission cost (and empty callhook data) for gateway router
+        bytes memory _data = abi.encode(maxSubmissionCost, bytes(""));
+
+        _updateDistribution(parentChainTokenAddr);
+        // As the caller of outboundTransferCustomRefund, this contract's alias is set as the callValueRfundAddress,
+        // given it affordance to cancel and receive callvalue refund. Since this contract can't call cancel, cancellation
+        // can't be performed. Generally calls to outboundTransferCustomRefund will create retryables with zero callValue
+        // (and even if there is callvalue, the refund would only take effect if the retryable expires).
+        parentChainGatewayRouter.outboundTransferCustomRefund{value: msg.value}({
+            _token: parentChainTokenAddr,
+            _refundTo: msg.sender, // send excess fees to the sender's address on the child chain
+            _to: destination,
+            _amount: amount,
+            _maxGas: gasLimit,
+            _gasPriceBid: maxFeePerGas,
+            _data: _data
+        });
+
+        emit FundsRouted(amount);
```

Comment:
It's the same event that gets emitted when routing Eth. Something like `event FundsRouted(address indexed token, uint256 amount)` would be more useful (also in the case of ChildToParent router)